pymatgen/core/structure.py

```diff
@@ -3124,14 +3124,18 @@ def merge_sites(self, tol=0.01, mode="sum"):
             inds = np.where(clusters == c)[0]
             species = self[inds[0]].species_and_occu
             coords = self[inds[0]].frac_coords
+            props = self[inds[0]].properties
             for n, i in enumerate(inds[1:]):
                 sp = self[i].species_and_occu
                 if mode == "s":
                     species += sp
                 offset = self[i].frac_coords - coords
                 coords += ((offset - np.round(offset)) / (n + 2)).astype(
                     coords.dtype)
-            sites.append(PeriodicSite(species, coords, self.lattice))
+                if self[i].properties != props and props[list(props.keys())[0]] is not None:
+                    props[list(props.keys())[0]] = None
+                    warnings.warn("Sites with different site_prop are merged. Site_prop is set to none")
```

Comment:
Can you change the warning to "Sites with different site property {} were merged, so property set to None." with the specific property name? As written, this doesn't seem like it'd support multiple site properties properly.